website/docs/reference/deploy/environment-import-export.mdx

```diff
@@ -0,0 +1,110 @@
+
+Additionally, these global configuration items are exported:
+* [custom context fields](../unleash-context.md#context-fields)
+* [feature tag types](../tags.md#tag-types)
+
+Importantly, while references to [segments](../segments.mdx)) are exported, the segments themselves are **not** exported. Consult the [import rejection](#import-rejection) section for more information.
+
+## Export
+
+You can export features either from a project search or the global feature search. Use the search functionality to narrow the results to the list of features you want to export and use the export button to select environment. All features included in your search results will be included in the export.
+
+<Figure caption="Search-driven export screen with WYSIWYG mode."  img="/img/export.png"/>
+
+## Import
+
+Import is a 3 stage process designed to be efficient and error-resistant.
+
+### Import stages
+
+* **upload** - you can upload previously exported JSON file or copy-paste export data from the exported JSON file into the code editor
+* **validation** - you will get feedback on any errors or warnings before you do the actual import. This makes sure your feature flags configurations. You will not be able to finish the import if you have errors. Warnings don't stop you from importing.
+* **import** - the actual import that creates a new configuration in the target environment or creates a [change request](../change-requests.md) when the environment has change requests enabled
+
+![Three-staged import process](/img/import.png)
+
+### Import rejection
+
+An import will be rejected when there's a possibility of data corruption.
+
+#### Context fields
+
+The exported and imported context fields have the same name but the target environment doesn't support some of the source environment context field values. If the context fields don't exist in the target environment they will be created.
+
+#### Segments
+
+The target Unleash instance doesn't have a matching segment. The actual constraints in the segment can be different. The matching is done by the segment name.
+
+#### Custom strategies
+
+The exported strategies include custom strategies that don't exist in the target Unleash instance. The matching is done by the strategy name.
+
+#### Existing features
+
+You're trying to import a feature that already exists in a different project on the target instance.
+
+#### Pending change request
+
+The target project and environment have a pending change request (applies to enterprise customers with change requests enabled for a given project and environment).
+
+### Import warnings
+
+Import validation will warn you when:
+* the features you are trying to import are already archived in the target instance in any project. Those features won't be imported again.
+* the strategies you are trying to import are custom and may have different [configuration parameters](../custom-activation-strategies.md#parameters). Those strategies will be imported anyway.
```

Comment:
So custom strategies always cause either a warning or an error, right? And why don't we validate custom strategies? Not worth the effort?

```suggestion
The import validation system will warn you about potential problems it finds in the import data. These warnings do not prevent you from importing the data, but you should read them carefully to ensure that Unleash does as you intend.

The following sections list things that the import tool will warn you about.

#### Archived features

The import tool will not import any features that have already been archived on the target Unleash instance. Because features are identified by their name, that means that if a feature called `feature-a` has been created and archived on the target Unleash instance, then a feature with the same name (`feature-a`) will not be imported.

If you permanently delete the archived `feature-a` from the target instance, then the new `feature-a` (in the import data) **will** be imported. 

#### Custom strategies

Unleash will verify that any custom strategies you are trying to import have already been defined on the target instance. However, it only does this verification by name. It does **not** validate that the definitions are otherwise the same or that they have the same [configuration parameters](../custom-activation-strategies.md#parameters). 

Before proceeding with the import, you should be sure that the custom strategy definitions are the same in the target instance and the imported custom strategies.
```

Reply:
Because you may want the custom strategy work differently in those two instances. So we leave it up to you to decide if you're ok exporting a custom strategy. Also my impression is that we may want to start using constraints instead of custom strategies. So not worth investing into it too much right now.

Reply:
Custom strategy gives a warning if you have it in a target instance. Custom strategy gives you an error if you don't have it in a target instance.

Reply:
Got it, thanks. In that case, I think the suggestion still stands, but we could maybe reword or remove the last section:

```suggestion
The import validation system will warn you about potential problems it finds in the import data. These warnings do not prevent you from importing the data, but you should read them carefully to ensure that Unleash does as you intend.

The following sections list things that the import tool will warn you about.

#### Archived features

The import tool will not import any features that have already been archived on the target Unleash instance. Because features are identified by their name, that means that if a feature called `feature-a` has been created and archived on the target Unleash instance, then a feature with the same name (`feature-a`) will not be imported.

If you permanently delete the archived `feature-a` from the target instance, then the new `feature-a` (in the import data) **will** be imported. 

#### Custom strategies

Unleash will verify that any custom strategies you are trying to import have already been defined on the target instance. However, it only does this verification by name. It does **not** validate that the definitions are otherwise the same or that they have the same [configuration parameters](../custom-activation-strategies.md#parameters). 
```